packages/app/components/token-picker/TokenPicker.tsx

```diff
@@ -4,51 +4,119 @@ import { useTokenListContext } from "@/context/TokenListContext";
 import { Token } from "@/models/token/types";
 import {
   BodyText,
+  Button,
   Icon,
   Modal,
   ModalBaseProps,
   ModalContent,
   ModalHeaderTitle,
   RadioButton
 } from "@/ui";
+import { ChangeEvent, RefObject, forwardRef, useEffect, useState } from "react";
 import { TokenIcon } from "../TokenIcon";
 import { TOKEN_PICKER_COMMON_TOKENS } from "./constants";
+import EmptyStateImg from "public/assets/images/empty-state-token-picker.svg";
 
+const HALF_SECOND = 500;
+
+interface TokenPickerProps extends ModalBaseProps {
+  initialFocusRef?: RefObject<HTMLInputElement>;
+}
+
+interface SearchBarProps {
+  onSearch: (event: ChangeEvent<HTMLInputElement>) => void;
+  value: string;
+}
 interface TokenListRowProps {
   token: Token;
   closeAction: () => void;
 }
 
 interface TokenListProps {
   closeAction: () => void;
+  onClearSearchClick: () => void;
   tokenList: Token[];
+  tokenSearchQuery?: string;
 }
 
-export const TokenPicker = ({ isOpen, closeAction }: ModalBaseProps) => {
+export const TokenPicker = ({
+  closeAction,
+  initialFocusRef,
+  isOpen
+}: TokenPickerProps) => {
+  const [tokenSearchQuery, setTokenSearchQuery] = useState("");
+  const [debouncedQuery, setDebouncedQuery] = useState(tokenSearchQuery);
+
   const { tokenList } = useTokenListContext();
 
+  const tokenListClearSearchCleanup = () => {
```

Comment:
I think If we remove the `clear` in the middle it be more clear. (pun intended).

I think CleanUp is already describing that, clear things and stuff.  It feels duplicate.

```suggestion
  const tokenListSearchCleanup = () => {
```